packages/shared/src/components/modals/EditWelcomePostModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { MouseEvent, ReactElement, useState } from 'react';
+import { useMutation, useQueryClient } from 'react-query';
+import { Modal } from './common/Modal';
+import { Button } from '../buttons/Button';
+import { NewCommentModalProps } from './NewCommentModal';
+import { PromptOptions, usePrompt } from '../../hooks/usePrompt';
+import PreviewTab from './PreviewTab';
+import { editPost, PostData } from '../../graphql/posts';
+import ContentWriteTab from './ContentWriteTab';
+import { getPostByIdKey } from '../../hooks/usePostById';
+import { useToastNotification } from '../../hooks/useToastNotification';
+
+enum EditPostTab {
+  Write = 'Write',
+  Preview = 'Preview',
+}
+
+const promptOptions: PromptOptions = {
+  title: 'Discard changes',
+  description: 'Are you sure you want to close and discard your changes?',
+  okButton: {
+    className: 'btn-primary-ketchup',
+    title: 'Discard',
+  },
+  cancelButton: {
+    title: 'Stay',
+  },
+};
+
+function EditWelcomePostModal({
+  post,
+  parentSelector,
+  onRequestClose,
+  ...props
+}: Omit<NewCommentModalProps, 'parentComment'>): ReactElement {
+  const client = useQueryClient();
+  const { displayToast } = useToastNotification();
+  const [input, setInput] = useState<string>(post.content);
+  const [errorMessage, setErrorMessage] = useState<string>(null);
+  const { mutateAsync: updatePost, isLoading } = useMutation(editPost, {
+    onSuccess: (updatedPost) => {
+      if (!updatedPost) return;
+
+      const key = getPostByIdKey(post.id);
+      client.setQueryData<PostData>(key, (data) =>
+        !data
+          ? { post: updatedPost }
+          : { post: { ...data.post, ...updatedPost } },
```

Comment:
Does edit mutation not return complete entity after db update? So we do not need this spread and we can just use `updatedPost`?

Reply:
They are two different queries, there could be a discrepancy with their properties defined on their own.

Reply:
@rebelchris just for useQuery, handlers are just fine and welcome for mutations.